src/System.Linq.Expressions/src/System/Linq/Expressions/IndexExpression.cs

```diff
@@ -383,52 +395,52 @@ private static void ValidateIndexedProperty(Expression instance, PropertyInfo in
                 ParameterInfo[] setParameters = setter.GetParametersCached();
                 if (setParameters.Length == 0)
                 {
-                    throw Error.SetterHasNoParams(nameof(indexer));
+                    throw Error.SetterHasNoParams(paramName);
                 }
 
                 // valueType is the type of the value passed to the setter (last parameter)
                 Type valueType = setParameters[setParameters.Length - 1].ParameterType;
                 if (valueType.IsByRef)
                 {
-                    throw Error.PropertyCannotHaveRefType(nameof(indexer));
+                    throw Error.PropertyCannotHaveRefType(paramName);
```

Comment:
If this branch wasn't here then anything that hit it would hit the `indexer.PropertyType != valueType` below instead. The result would be the same exception type and same parameter name, but a different (also valid) error message. Could it be removed?